### PR TITLE
Add K8s Autosceduling output node

### PR DESCRIPTION
**Pull request by nathanielc**
Adds a `k8sAutoscale` node that can change the replica counts on k8s scale resources based on dynamic criteria.

- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Ensure the credentials work both inside and outside a k8s cluster.

### CODE REVIEW

File: pipeline/k8s_autoscale.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+//             // Get the name of the deployment from the 'deployment' tag.
+//             .resourceNameTag('deployment')
+//             .min(min)
+//             .max(max)
+//             // Set the desired number of replicas based on target.
+//             .replicas(lambda: int(ceil("total_requests" / target)))
+//         |influxDBOut()
+//             .database('deployments')
+//             .measurement('scale_events')
+//             .precision('s')
```
Comment (nathanielc) [inline pipeline/k8s_autoscale.go]: @goller Does this TICkscript make sense to you, specifically the `|k8sAutoscale()` part?

Want to make sure this API fits what you would expect using k8s.

Comment (goller) [inline pipeline/k8s_autoscale.go]: @nathanielc Looks wonderful!

**Comment by nathanielc** [comment]
@joelegasse This is ready for review.

**Comment by joelegasse** [comment]
What would you think about keeping the `k8s/client` types in the `k8s` package? Most of your usage aliases the package to `k8s` anyway. And when you're not doing that, you have a stutter: `client.Client`

File: k8s_autoscale.go [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+	// Update local copy of state
+	change := e.New - e.Old
+	state.current = e.New
+
+	// Check last change cooldown times
+	var counter *expvar.Int
+	switch {
+	case change > 0:
+		if t.Before(state.lastIncrease.Add(k.k.IncreaseCooldown)) {
+			// Still hot, nothing to do
```
Comment (nathanielc) [inline k8s_autoscale.go]: Add stat for dropped hot events.

**Pull request merged**